### ISSUE TITLE
core data types: ByteString test / interop

### ISSUE THREAD

**Issue by dckc** [ocapn/ocapn#48]
Inclusion of the datatype seems to have consensus, but Agoric lacks support, so we can't demonstrate interop yet.

A test that all 3 of Spritely, Agoric, and capnp pass would be great.

ref @erighs https://github.com/ocapn/ocapn/issues/5#issuecomment-1550492042
> I believe Agoric's current implementations are already in complete conformance with everything but the following:
>
> - Agoric's complete absence of ByteString support

**Comment by erights** [comment]
Above I'm quoted as saying
> > I believe Agoric's current implementations are already in complete conformance with everything but the following:
> >
> > - Agoric's complete absence of ByteString support

Since then, at
https://github.com/endojs/endo/issues/1587
I have identified more conformance issues for Agoric. Five of these are still currently unchecked, including this absence of ByteString support.

**Comment by kriskowal** [comment]
Please join me in very seriously expressing a preference for one of the following in reactji.

- byte array 🚀
- byte vector 🎉
- byte string 👀

**Comment by kriskowal** [comment]
@jar398 Let’s put a record of results on the agenda for next meeting.

**Comment by dckc** [comment]
Can we please postpone choosing a name until we have a test case demonstrating interoperability?

**Comment by erights** [comment]
> Can we please postpone choosing a name until we have a test case demonstrating interoperability?

Just saw this. At the OCapN meeting just now, we did go with the poll winner, "byte array". I would prefer not to revisit this, but to accept it as the winner.

How might getting to an interoperable test case affect our name choice?

**Comment by dckc** [comment]
> How might getting to an interoperable test case affect our name choice?

I can join you in hoping that it doesn't.

I would very much like to establish a pattern where we have a test case in place when we decide issues, though. So I suppose as long as noone aims to close this issue before we have a test, I'm content.

**Comment by kriskowal** [comment]
I like tests and agree we should settle many issues only after we have tested code on both the Agoric and Spritely platforms. I do not understand how it would be mechanically possible for the name of an entity in the language-agnostic model could cause a concrete test to fail. I however can see how a cross-language test might reveal that the type should not exist in the data model at all.
